packages/epics/src/agreements/components/agreement-form-payment-schedule.tsx

```diff
@@ -0,0 +1,257 @@
+'use client';
+
+import { useEffect } from 'react';
+import {
+  useForm,
+  Controller,
+  useFieldArray,
+  FormProvider,
+} from 'react-hook-form';
+import {
+  DropdownMenu,
+  DropdownMenuTrigger,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  Button,
+  Label,
+  Input,
+  DatePicker,
+} from '@hypha-platform/ui';
+import { cn } from '@hypha-platform/lib/utils';
+import { ChevronDownIcon, PlusIcon, Cross2Icon } from '@radix-ui/react-icons';
+import { PercentIcon } from 'lucide-react';
+import { isBefore } from 'date-fns';
+
+const options = ['Immediately', 'Future Payment', 'Milestones'] as const;
+type Option = (typeof options)[number];
+
+type DateRange = {
+  from: Date | undefined;
+  to: Date | undefined;
+};
+
+type Milestone = {
+  percentage: string;
+  dateRange: DateRange;
+};
+
+type FormValues = {
+  option: Option;
+  futureDate?: Date;
+  milestones: Milestone[];
+};
+
+type AgreementFormPaymentScheduleProps = {
+  onChange?: (value: FormValues) => void;
+};
+
+export function AgreementFormPaymentSchedule({
+  onChange,
+}: AgreementFormPaymentScheduleProps) {
+  const methods = useForm<FormValues>({
+    defaultValues: {
+      option: 'Immediately',
+      futureDate: undefined,
+      milestones: [],
+    },
+    mode: 'onChange',
+  });
```

Comment:
I think we should check [`useFormContext`](https://react-hook-form.com/docs/useformcontext) to make the form composable